shell/common/api/electron_api_native_image_mac.mm

```diff
@@ -34,6 +39,49 @@
   return def;
 }
 
+// static
+v8::Local<v8::Promise> NativeImage::CreateThumbnailFromPath(
+    v8::Isolate* isolate,
+    const base::FilePath& path,
+    const gfx::Size& size) {
+  gin_helper::Promise<gfx::Image> promise(isolate);
+  v8::Local<v8::Promise> handle = promise.GetHandle();
+
+  if (size.IsEmpty()) {
+    promise.RejectWithErrorMessage("size must not be empty");
+    return handle;
+  }
+
+  CGSize cg_size = size.ToCGSize();
+  base::ScopedCFTypeRef<CFURLRef> cfurl = base::mac::FilePathToCFURL(path);
+  base::ScopedCFTypeRef<QLThumbnailRef> ql_thumbnail(
+      QLThumbnailCreate(kCFAllocatorDefault, cfurl, cg_size, NULL));
```

Comment:
This function is deprecated. Is it the best option?

Reply:
I believe the other option is QLTHumbnailImageCreate (https://developer.apple.com/documentation/quicklook/1402623-qlthumbnailimagecreate?language=objc) 

but @MarshallOfSound mentioned that we don't want to use that as it is a synchronous function